### DISCUSSION TITLE
How do I deploy an ECS Fargate service with an ALB?

### DISCUSSION THREAD

**Discussion by yorinasub17** [gruntwork-io/knowledge-base#168] [Help] [answered]
I'm trying to configure the [ecs-service module](https://github.com/gruntwork-io/terraform-aws-ecs/tree/master/modules/ecs-service) in Fargate mode that is linked to an ALB. I have figured out the variable configuration for setting up the service to deploy in Fargate mode, but I am struggling to figure out what variables I need to configure to link it to an ALB. Which variables should I configure?

NOTE: I already tried copying the `ecs-service` module block from the [docker-fargate-service-with-alb](https://github.com/gruntwork-io/terraform-aws-ecs/tree/master/examples/docker-fargate-service-with-alb), but I am getting the following error:

```
Error: failed creating ECS service (SERVICE_NAME): InvalidParameterException: The target group with 
targetGroupArn arn:aws:elasticloadbalancing:eu-west-1:xxxxxxxxx:targetgroup/SERVICE_NAME/c5717e184d65dcf7
does not have an associated load balancer
```

**Comment by denis256** [comment, marked as answer]
Hi, without `ecs-service` definition is not clear how which variables got passed to module, but should be set [`elb_target_groups`](https://github.com/gruntwork-io/terraform-aws-ecs/blob/master/examples/docker-fargate-service-with-alb/main.tf#L68) on `ecs-service` which should reference [`alb`](https://github.com/gruntwork-io/terraform-aws-ecs/blob/master/examples/docker-fargate-service-with-alb/main.tf#L155)

**Reply by denis256** [reply]
Additionally, found that a similar error occurs if there is no `aws_alb_listener_rule` defined.

Example definition can be found in:
https://github.com/gruntwork-io/terraform-aws-ecs/blob/master/examples/docker-fargate-service-with-alb/main.tf#L251

In my tests, I commented `aws_alb_listener_rule` from `docker-fargate-service-with-alb` and get the same error during apply
